docs/_developer_onboarding.md

```diff
@@ -117,41 +117,42 @@ You can also have access to the aars through the [release](https://github.com/ma
-
-Checkout both [matrix-rust-sdk](https://github.com/matrix-org/matrix-rust-sdk) and [matrix-rust-components-kotlin](https://github.com/matrix-org/matrix-rust-components-kotlin) repositories
-```shell
-git clone git@github.com:matrix-org/matrix-rust-sdk.git
-git clone git@github.com:matrix-org/matrix-rust-components-kotlin.git
-```
-
-Then you can launch the build script from the matrix-rust-components-kotlin repository with the following params:
-
-- `-p` Local path to the rust-sdk repository
-- `-o` Optional output path with the expected name of the aar file. By default the aar will be located in the corresponding build/outputs/aar directory.
-- `-r` Flag to build in release mode
-- `-m` Option to select the gradle module to build. Default is sdk.
-- `-t` Option to to select an android target to build against. Default will build for all targets.
-
-So for example to build the sdk against aarch64-linux-android target and copy the generated aar to Element X project:
-
-```shell
-./scripts/build.sh -p [YOUR MATRIX RUST SDK PATH] -t aarch64-linux-android -o [YOUR element-x-android PATH]/libraries/rustsdk/matrix-rust-sdk.aar
-```
+Prerequisites:
+* Install the Android NDK (Native Development Kit). To do this from within
+  Android Studio:
+    1. **Tools > SDK Manager**
+    2. Click the **SDK Tools** tab.
+    3. Select the **NDK (Side by side)** checkbox
+    4. Click **OK**.
+    5. Click **OK**.
+    6. When the installation is complete, click **Finish**.
+* Install `cargo-ndk`:
+  ```
+  cargo install cargo-ndk
+  ```
+* Install the Android Rust toolchain:
+  ```
+  rustup target add aarch64-linux-android
+  ```
+* Depending on the location of the Android SDK, you may need to set
+  `ANDROID_HOME`:
+  ```
+  export ANDROID_HOME=$HOME/android/sdk
+  ```
+
+You can then build the Rust SDK by running the script
+[`tools/sdk/build_rust_sdk.sh`](../tools/sdk/build_rust_sdk.sh) and just answering
+the questions.
+
+This will prompt you for the path to the Rust SDK, then build it and
+`matrix-rust-components-kotlin`, eventually producing an aar file at
+`./libraries/rustsdk/matrix-rust-sdk.aar`, which will be picked up
+automatically by the Element X Android build.
 
 Troubleshooting:
  - You may need to set `ANDROID_NDK_HOME` e.g `export ANDROID_NDK_HOME=~/Library/Android/sdk/ndk`.
  - If you get the error `thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', .cargo/registry/src/index.crates.io-6f17d22bba15001f/cargo-ndk-2.11.0/src/cli.rs:345:18` try updating your Cargo NDK version. In this case, 2.11.0 is too old so `cargo install cargo-ndk` to install a newer version.
- - If you get the error `Unsupported class file major version 64` try changing your JVM version. In this case, Java 20 is not supported in Gradle yet, so downgrade to an earlier version (Java 17 worked in this case).
```

Comment:
Empirically, Gradle now supports Java 21

Reply:
fair enough. How does one go about changing the JVM version used in those two separate places?

Presumably one of them is "Android Studio | File | Settings | Build, Execution, Deployment | Build Tools | Gradle | Gradle JDK", but which? And how do you change the other?

Reply:
If you used the command line for both I think you'd just have to change your `JAVA_HOME` env var to point at the SDK version.

If you're building the app using Android Studio though, you'd also have to change the JDK used as you suggested, if I'm not mistaken.

Reply:
Ok, so how about:

```suggestion
 - If you get the error `Unsupported class file major version <n>`, try changing your JVM version by setting
   `JAVA_HOME` and, if building via Android Studio, "File | Settings | Build, Execution, Deployment | Build Tools | Gradle | Gradle JDK".
```